srcjs/output_binding_image.js

```diff
@@ -312,90 +358,112 @@ imageutils.initCoordmap = function($el, coordmap) {
+      const prefix = key.substring(0, 1);
+      
+      if (prefix === "x") {
+        return offset_css[key] / pixel_scaling.x;
+      } else if (prefix === "y") {
+        return offset_css[key] / pixel_scaling.y;
+      }
+      return null;
+    });
 
-  // Shift an array of values so that they are within a min and max.
-  // The vals will be shifted so that they maintain the same spacing
-  // internally. If the range in vals is larger than the range of
-  // min and max, the result might not make sense.
-  coordmap.shiftToRange = function(vals, min, max) {
-    if (!(vals instanceof Array))
-      vals = [vals];
-
-    var maxval = Math.max.apply(null, vals);
-    var minval = Math.min.apply(null, vals);
-    var shiftAmount = 0;
-    if (maxval > max) {
-      shiftAmount = max - maxval;
-    } else if (minval < min) {
-      shiftAmount = min - minval;
-    }
+    return result;
+  };
 
-    var newvals = [];
-    for (var i=0; i<vals.length; i++) {
-      newvals[i] = vals[i] + shiftAmount;
-    }
-    return newvals;
+  // Given an offset in an img, in source image pixels, return the
+  // corresponding offset in CSS pixels. If the img content is 1000 pixels
+  // wide, but is scaled to 400 pixels on screen, and the input is x:1000,
+  // then this will return x:400.
+  coordmap.scaleImgToCss = function(offset_img) {
+    const pixel_scaling = findImgToCssScalingRatio($img);
+
+    const result = mapValues(offset_img, (value, key) => {
+      const prefix = key.substring(0, 1);
+      
+      if (prefix === "x") {
+        return offset_img[key] * pixel_scaling.x;
+      } else if (prefix === "y") {
+        return offset_img[key] * pixel_scaling.y;
+      }
+      return null;
+    });
+
+    return result;
   };
 
-  // Given an offset, return an object representing which panel it's in. The
-  // `expand` argument tells it to expand the panel area by that many pixels.
-  // It's possible for an offset to be within more than one panel, because
-  // of the `expand` value. If that's the case, find the nearest panel.
-  coordmap.getPanel = function(offset, expand) {
-    expand = expand || 0;
+  coordmap.cssToImgScalingRatio = function() {
+    const res = findImgToCssScalingRatio($img);
+    return {
+      x: 1 / res.x,
+      y: 1 / res.y
+    };
+  };
 
-    var x = offset.x;
-    var y = offset.y;
+  // Given an offset in css pixels, return an object representing which panel
+  // it's in. The `expand` argument tells it to expand the panel area by that
+  // many pixels. It's possible for an offset to be within more than one
+  // panel, because of the `expand` value. If that's the case, find the
+  // nearest panel.
+  coordmap.getPanelCss = function(offset_css, expand = 0) {
+    const offset_img = coordmap.scaleCssToImg(offset_css);
+    const x = offset_img.x;
+    const y = offset_img.y;
+
+    // Convert expand from css pixels to img pixels
+    const imgToCssRatio = findImgToCssScalingRatio($img);
```

Comment:
Reverse this--`const cssToImgRatio = coordmap.cssToImgScalingRatio();` and change division below to multiplication